Start of porting bug fixes

FireBurn opened the pull request:
This is my first attempt

These patches have been compile tested, and I've set up an OpenAM instance using my scripts

They haven't been tested further than that I'd be grateful if you could take a look

If you're happy with them I'll work on some more of them tomorrow

FireBurn commented:
I'm not sure my take on OPENAM-8643 fixes the issue

FireBurn commented:
OPENAM-7583 has been the most awkward one so far, finally got it compiling and running a simple instance

aldaris commented:
OPENAM-7958 looks good too

aldaris commented:
OPENAM-8106 looks good.

aldaris commented:
OPENAM-6562 and OPENAM-4743 looks good.

aldaris commented:
OPENAM-8321 and OPENAM-8248 look good too.

aldaris commented:
OPENAM-8951 and OPENAM-9216 are good too.

aldaris commented on openam-core/src/main/java/com/sun/identity/idm/remote/IdRemoteServicesImpl.java:
```diff
@@ -38,6 +36,7 @@
 import com.iplanet.am.util.SystemProperties;
 import com.iplanet.sso.SSOException;
 import com.iplanet.sso.SSOToken;
+import com.sun.identity.authentication.spi.AuthLoginException;
```
unused import

aldaris commented:
OPENAM-8258 looks good

aldaris commented on openam-forgerock-rest/src/main/java/org/forgerock/openam/forgerockrest/IdentityResource.java:
```diff
@@ -644,21 +645,19 @@ private void anonymousUpdate(final ServerContext context, final ActionRequest re
+            if (newPassword == null || newPassword.isEmpty()) {
                 throw new BadRequestException("new password not provided");
             }
 
```
This fix is not complete, https://github.com/aldaris/OpenAM/commit/d08466f8cb67b699ebcfb6052b348f7b7fc5c00b#diff-343b84b79e4cbc68ed84eccc7a80fc1fR830 plays key role in the fix.

FireBurn replied:
Is it the "json(object(field(USERNAME, username), field(USER_PASSWORD, newPassword))" that's important or the "thenAsync"?

aldaris replied:
former

FireBurn commented:
I think I've fixed up OPENAM-6373, added in the line and the extra bits required to make it work

I don't think there was anything wrong with OPENAM-7063 and I've slotted in OPENAM-8117 after it

Lastly I removed the extra import from OPENAM-7938

FireBurn commented:
I think I fixed up OPENAM-8643, it's compiling now at least

aldaris commented:
This review is getting a bit unmanageable, can we divide this up to smaller pieces of work?

FireBurn commented:
To implement OPENAM-7362 I had to convert getAllServerIDs to be a Collection<String> 

Everything compiled OK, but I'm not sure if this will affect where it's used else where

FireBurn commented:
If I close this PR, create a new one with the ones you've already OK'd then create seperate pull requests for each OPENAM fix?

aldaris commented:
That's one way to go about it, but I fear some of these fixes may have dependencies inbetween. I'm happy to review 3-4 commits at a time in quick succession if that helps.

FireBurn commented:
Let me know, how you'd like things split up, and I'll get it sorted

FireBurn commented:
Closing this and starting again
